integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+		)
+		require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
+		endSeqNum[sourceChain1] = startSeqNum[sourceChain1] + ccipocr3.SeqNum(numMessages) - 1
+	})
+
+	t.Run("batch data only messages from multiple sources", func(t *testing.T) {
+		var (
+			wg   sync.WaitGroup
+			mx   sync.Mutex
+			errs []error
+		)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain1)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain2)
+
+		wg.Wait()
+
+		for _, err := range errs {
+			require.NoError(t, err)
+		}
+
+		// confirm the commit reports
+		var (
+			sourceChain1Report *offramp.OffRampCommitReportAccepted
+			sourceChain2Report *offramp.OffRampCommitReportAccepted
+		)
+		errs = nil
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain1Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
+				e.Env.Chains[sourceChain1],
+				e.Env.Chains[destChain],
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+			)
+
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain2Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
+				e.Env.Chains[sourceChain2],
+				e.Env.Chains[destChain],
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain2], endSeqNum[sourceChain2]),
+			)
+
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}()
+
+		t.Log("waiting for commit report")
+		wg.Wait()
+
+		for _, err := range errs {
+			require.NoError(t, err)
+		}
+
+		// the reports should be the same for both, since both roots should be batched within
+		// that one report.
+		require.Equal(t, sourceChain1Report, sourceChain2Report, "commit reports should be the same")
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
```

Comment:
It's not used anywhere, is it?

Reply:
This isn't used anywhere but if we add test cases further down (which was the plan, wanted to add something where we manual exec a message that is part of a root with multiple msgs)

Reply:
If it's not gonna be used in this PR then better remove them to make the next PR easier to understand when seeing it added.